services/galley/src/Galley/External/LegalHoldService.hs

```diff
@@ -0,0 +1,228 @@
+    , removeLegalHold
+
+      -- * helpers
+    , validateServiceKey
+
+      -- * types
+    , OpaqueAuthToken(..)
+    ) where
+
+import Imports
+import Data.Id
+import Galley.API.Error
+import Galley.App
+import Bilge.Retry
+import Bilge.Response
+import Brig.Types.Provider
+import Brig.Types.Team.LegalHold
+import Control.Exception.Enclosed (handleAny)
+import Control.Lens hiding ((.=), (#))
+import Control.Monad.Catch
+import Control.Retry
+import Data.Aeson
+import Data.ByteString.Conversion.To
+import Data.Misc
+import Network.HTTP.Types
+import Ssl.Util
+import URI.ByteString (uriPath)
+
+import qualified Bilge
+import qualified Galley.Data.LegalHold      as LegalHoldData
+import qualified Data.ByteString.Lazy.Char8 as LC8
+import qualified Data.ByteString            as BS
+import qualified Network.HTTP.Client        as Http
+import qualified OpenSSL.EVP.Digest         as SSL
+import qualified OpenSSL.EVP.PKey           as SSL
+import qualified OpenSSL.PEM                as SSL
+import qualified OpenSSL.RSA                as SSL
+import qualified OpenSSL.Session            as SSL
+import qualified Ssl.Util                   as SSL
+import qualified System.Logger.Class        as Log
+
+----------------------------------------------------------------------
+-- api
+
+-- | Get /status from legal hold service; throw 'Wai.Error' if things go wrong.
+checkLegalHoldServiceStatus :: Fingerprint Rsa -> HttpsUrl -> Galley ()
+checkLegalHoldServiceStatus fpr url = do
+    resp <- makeVerifiedRequestFreshManager fpr url reqBuilder
+    if | Bilge.statusCode resp < 400 -> pure ()
+       | otherwise -> do
+            Log.info . Log.msg $ showResponse resp
+            throwM legalHoldServiceBadResponse
+  where
+    reqBuilder :: Http.Request -> Http.Request
+    reqBuilder
+        = Bilge.paths ["status"]
+        . Bilge.method GET
+        . Bilge.expect2xx
+
+-- | @POST /initiate@.
+requestNewDevice :: TeamId -> UserId -> Galley NewLegalHoldClient
+requestNewDevice tid uid = do
+    resp <- makeLegalHoldServiceRequest tid reqParams
+    case eitherDecode (responseBody resp) of
+        Left e -> do
+            Log.info . Log.msg $ "Error decoding NewLegalHoldClient: " <> e
+            throwM legalHoldServiceBadResponse
+        Right client -> pure client
+  where
+    reqParams =
+        Bilge.paths ["initiate"]
+      . Bilge.json (RequestNewLegalHoldClient uid tid)
+      . Bilge.method POST
+      . Bilge.acceptJson
+      . Bilge.expect2xx
+
+-- | @POST /confirm@
+-- Confirm that a device has been linked to a user and provide an authorization token
+confirmLegalHold :: ClientId
+                 -> TeamId
+                 -> UserId
+                 -> OpaqueAuthToken -- ^ TODO: Replace with 'LegalHold' token type
+                 -> Galley ()
+confirmLegalHold clientId tid uid legalHoldAuthToken = do
+    void $ makeLegalHoldServiceRequest tid reqParams
+  where
+    reqParams =
+        Bilge.paths ["confirm"]
```

Comment:
We should definitely fix this before merging; this path should be appended to `legalHoldServiceUrl`

Reply:
isn't this already happening?

- https://github.com/wireapp/wire-server/blob/5325b1350534f1069cd07fcb893e78873a54d91c/services/galley/test/integration/API/Teams/LegalHold.hs#L743
- https://github.com/wireapp/wire-server/blob/5325b1350534f1069cd07fcb893e78873a54d91c/services/galley/test/integration/API/Teams/LegalHold.hs#L780-L786
- https://github.com/wireapp/wire-server/blob/5325b1350534f1069cd07fcb893e78873a54d91c/services/galley/src/Galley/External/LegalHoldService.hs#L176

Reply:
Oh so the comment was outdated?

Reply:
Seems so, yes.  Sorry about that.

Reply:
(I only saw the comment after I answered to your answer to it.  I think your answer was to the line above the comment, so it didn't show in the email.  Life is full of tricks!)